### PR TITLE
Fix deprecated gradle API usage

### PR DESCRIPTION
**Pull request by wolfs**
This fixes the test problems of https://github.com/JetBrains/gradle-idea-ext-plugin/pull/82.

### CODE REVIEW

File: src/test/kotlin/org/jetbrains/gradle/ext/BuildIdeArtifactTest.kt [hunk 1]
```diff
@@ -5,343 +5,362 @@ import junit.framework.Assert.assertFalse
...
+    }
 
-    } finally {
-      temp.deleteRecursively()
+    fun createTestArchive(sourceDir: File, destinationFile: File) {
+        var out = ZipOutputStream(BufferedOutputStream(FileOutputStream(destinationFile)))
+        out.use { usedZipStream ->
+            sourceDir.walkBottomUp().forEach { file ->
+                if (!file.isDirectory()) {
+                    var entry = ZipEntry(file.absolutePath.removePrefix(sourceDir.absolutePath + "/"))
```
Comment (nskvortsov) [inline src/test/kotlin/org/jetbrains/gradle/ext/BuildIdeArtifactTest.kt]: Please, replace `sourceDir.absolutePath + "/"` with `sourceDir.absolutePath + File.separator` to fix test problem on Windows

Comment (wolfs) [reply src/test/kotlin/org/jetbrains/gradle/ext/BuildIdeArtifactTest.kt]: Fixed.

**Comment by wolfs** [comment]
I don't understand why the plugin is creating zip files in its production code: https://github.com/JetBrains/gradle-idea-ext-plugin/blob/9b54b63f296f4a87f5f5e3193d57fc7401212fb3/src/main/groovy/org/jetbrains/gradle/ext/IdeArtifacts.groovy#L171

Isn't this plugin only about IDEA configuration?

**Comment by nskvortsov** [comment]
> 
> 
> I don't understand why the plugin is creating zip files in its production code:
> 
> https://github.com/JetBrains/gradle-idea-ext-plugin/blob/9b54b63f296f4a87f5f5e3193d57fc7401212fb3/src/main/groovy/org/jetbrains/gradle/ext/IdeArtifacts.groovy#L171
> 
> Isn't this plugin only about IDEA configuration?

This particular feature (creating idea artifacts) was requested by our colleagues from Kotlin project. It includes also a task `org.jetbrains.gradle.ext.BuildIdeArtifact` that allows building such artifacts from Gradle CLI

**Pull request merged**